### ISSUE TITLE
HiGHS crash while adding Clique 

### ISSUE THREAD

**Issue by Dexterux** [ERGO-Code/HiGHS#1372]
Hi everyone,

I'm testing HiGHS and I encountered a crash while solving a problem on Windows with HiGHS v1.5.3, v1.5.1 and the code from branch master.

The output of the solver is the following:
```
Running HiGHS 1.5.3 [date: 2023-06-28, git hash: a29e36f11-dirty]
Copyright (c) 2023 HiGHS under MIT licence terms
Rows:        6713 lower bounds exceeding       -1e+20 are treated as -Infinity
Rows:         217 upper bounds exceeding        1e+20 are treated as +Infinity

Presolve is switched off
Objective function is integral with scale 1

Solving MIP model with:
   18772 rows
   47675 cols (47418 binary, 257 integer, 0 implied int., 0 continuous)
   2348243 nonzeros

        Nodes      |    B&B Tree     |            Objective Bounds              |  Dynamic Constraints |       Work      
     Proc. InQueue |  Leaves   Expl. | BestBound       BestSol              Gap |   Cuts   InLp Confl. | LpIters     Time

         0       0         0   0.00%   0               inf                  inf        0      0      0         0     1.0s
         0       0         0   0.00%   896.8010189     inf                  inf        0      0      2      1241     9.6s
         0       0         0   0.00%   996.3344742     inf                  inf     1037     56      6      1429    17.1s
         0       0         0   0.00%   1014.008907     inf                  inf     1762     97      8      1651    24.6s
         0       0         0   0.00%   1028.754132     inf                  inf     2304    128     10      1833    29.8s
 C       0       0         0   0.00%   1076.135025     16640             93.53%     2545    170     24      2041    62.8s
         0       0         0   0.00%   1130.986398     16640             93.20%     2919    292     24      2792    69.0s
         0       0         0   0.00%   1218.879657     16640             92.68%     3390    344     24      4407    74.9s
         0       0         0   0.00%   1232.826308     16640             92.59%     3984    395     24      5106    80.9s
         0       0         0   0.00%   1237.830329     16640             92.56%     4490    358     24      5626    86.1s
         0       0         0   0.00%   1247.74377      16640             92.50%     4542    326     24      6522    92.1s
 L       0       0         0   0.00%   1248.171066     1507              17.18%     3753    344     24      6714   171.4s
 L       0       0         0   0.00%   1259.645304     1505              16.30%     3664    302     24      7883   190.8s

Symmetry detection completed in 0.9s
Found 2 full orbitope(s) acting on 12 columns

```
No error is given, and the solver crashes. 

Unfortunately the problem is too big, and so far I was not able to run it with a debugger, but from the stack trace is seems that the problem occur in the method doAddClique

![image](https://github.com/ERGO-Code/HiGHS/assets/3617112/56aceae8-ea8c-4fda-a2ba-f23bb2560742)

To reproduce the bug I had to save the vectors of coefficients in files and load them to rebuild the model, since saving the model and reading it from a MPS file gave different results and didn't end with the crash.

I attach a CMake project containing the source code that may help to reproduce the bug, containing a `main.cpp` file and the text files with the coefficients for the different part of the model. I built both the test code and the HiGHS library with Visual Studio 2019

[HiGHS_test.zip](https://github.com/ERGO-Code/HiGHS/files/12217133/HiGHS_test.zip)

**Comment by jajhall** [comment]
Oof, that's an awkward one to look at, but I'll try to reproduce it!

**Comment by lkapelevich** [comment]
> Unfortunately that's the only OS on which I can debug code.

because of Visual Studio issues, or something else?

**Comment by jajhall** [comment]
I've used Visual Studio a few times in the past to debug Windows issues, but every other time I try to use it it fails to build HiGHS - presumably because it's been updated in the interim, or some other fickle Windows issue.

I should add that @galabovaa works on a Mac, so we've got that environment covered, too.

**Comment by lkapelevich** [comment]
Weird. In case it's helpful- I _can_ build your latest branch (and debug with some tweaks) in Visual Studio.
My output starts to deviate a bit from the screenshot in this post though.

**Comment by lkapelevich** [comment]
I can reproduce, but only with release builds so far. There is something weird with `InnerLeaf::capacity()`.

**Comment by Dexterux** [comment]
Sorry, I wasn't able to produce meaningful results while running with a debugger: after a day waiting for the crash I had to kill the process

**Comment by fwesselm** [comment]
Hi!

Thanks to @lkapelevich who has spent more time debugging this, I may have found a bug in HighsHashTree.h. However, this does not fix the issue. Namely, there is a line:

memset(sizes, 0, branchSize);

I think the intention here is to zero out "branchSize" elements and not bytes. Therefore, I think it should be

memset(sizes, 0, branchSize * sizeof(uint8_t));

**Comment by lkapelevich** [comment]
Smaller code to get a crash on Windows: https://gist.github.com/lkapelevich/49d34ed9c96711e6b409a91d4a190531
Hopefully this is debuggable on Linux too, I can see that in [this ](https://github.com/ERGO-Code/HiGHS/blob/master/src/util/HighsHashTree.h#L854) switch statement none of the cases are hit and there is no default.

**Comment by fwesselm** [comment]
Hello! 
[hashTreeBug.txt](https://github.com/ERGO-Code/HiGHS/files/12598779/hashTreeBug.txt)
I have tried to debug this a little more. I think there is a bug in HighsHashTree.h, namely in the handling of the "hashes" array.

The code is currently using "memmove" which causes the incorrect behavior, I think (- it produces duplicate elements in the "hashes" array for the smaller reproduction example found by @lkapelevich). I have changed the code to use "std::move_backward", similar to the "entries" array (see attached diff). I am not sure if "memmove" was used for efficiency or some other reason.

It seems that HiGHS can now solve the reproduction example to optimality. However, I have to run some more MIPs to see if there are no undesired side effects.
